### PR TITLE
Added committee.ConsumeQueue filtering

### PR DESCRIPTION
**Pull request by AKorpusenko**
## Description

Committee queue currently has disabled filtering on `ConsumeQueue` method. Inconsistent order of incoming p2p messages causing errors on stage. Bringing the filter back adding new priorizer for committee queue

## Changes for committee messages prioritizer

Slot order, older slot first

###  Order of priority

- Post-consensus

   - Commit

   - Round change

   - Proposal

   - Prepare

### CODE REVIEW

File: protocol/v2/ssv/validator/committee_queue.go [hunk 1]
```diff
@@ -99,50 +101,50 @@ func (v *Committee) ConsumeQueue(logger *zap.Logger, slot phase0.Slot, handler M
...
-		// 	}
-
-		// 	if sm.Height != state.Height || sm.Round != state.Round {
-		// 		return true
-		// 	}
-		// 	return sm.MsgType != specqbft.PrepareMsgType && sm.MsgType != specqbft.CommitMsgType
-		// }
+		runner := v.Runners[slot]
+		if runner == nil {
+			return fmt.Errorf("could not get duty runner for slot", slot)
```
[the remark this answers is not in the document]
Comment (AKorpusenko) [reply protocol/v2/ssv/validator/committee_queue.go]: Thanks!

File: protocol/v2/ssv/validator/committee_queue.go [hunk 2]
```diff
@@ -99,50 +101,53 @@ func (v *Committee) ConsumeQueue(logger *zap.Logger, slot phase0.Slot, handler M
...
-		// 	}
-
-		// 	if sm.Height != state.Height || sm.Round != state.Round {
-		// 		return true
-		// 	}
-		// 	return sm.MsgType != specqbft.PrepareMsgType && sm.MsgType != specqbft.CommitMsgType
-		// }
+		v.mtx.Lock()
+		runner := v.Runners[slot]
+		v.mtx.Unlock()
```
Comment (y0sher) [inline protocol/v2/ssv/validator/committee_queue.go:106]: There is no point in getting this from the map with the mutex each iteration of the loop. either pass it down from `StartDuty` into `ConsumeQueue` (better solution imo) or get it once before the `for` loop.

File: protocol/v2/ssv/validator/timer.go [hunk 3]
```diff
@@ -86,8 +86,8 @@ func (v *Committee) onTimeout(logger *zap.Logger, identifier spectypes.MessageID
 		//if v.state != uint32(Started) {
 		//	return
 		//}
-
 		dr := v.Runners[phase0.Slot(height)]
+
```
Comment (y0sher) [inline protocol/v2/ssv/validator/timer.go:90]: ?

**Review by y0sher — CHANGES_REQUESTED** [review]
check my comments please

File: protocol/v2/ssv/validator/committee_queue.go [hunk 4]
```diff
@@ -99,50 +101,53 @@ func (v *Committee) ConsumeQueue(logger *zap.Logger, slot phase0.Slot, handler M
...
+		var runningInstance *instance.Instance
+		if runner.HasRunningDuty() {
+			runningInstance = runner.GetBaseRunner().State.RunningInstance
+			if runningInstance != nil {
+				decided, _ := runningInstance.IsDecided()
+				state.HasRunningInstance = !decided
+			}
+		}
+
+		state.Slot = slot
```
[the remark this answers is not in the document]
Comment (y0sher) [reply protocol/v2/ssv/validator/committee_queue.go]: @AKorpusenko looked like you skipped this comment

File: protocol/v2/ssv/validator/committee_queue.go [hunk 5]
```diff
@@ -90,59 +98,53 @@ func (v *Committee) ConsumeQueue(logger *zap.Logger, slot phase0.Slot, handler M
 	if err != nil {
 		return err
 	}
+	
+	state := *q.queueState
+	state.Slot = slot
```
Comment (y0sher) [inline protocol/v2/ssv/validator/committee_queue.go:103]: I still don't get why this is here and not in construction (StartDuty) where the queue is created. (its 0 there). Don't forget to add it in the new HandleMessage where we create the queue if we don't have it yet on a new msg.

**Review by y0sher — CHANGES_REQUESTED** [review]

File: protocol/v2/ssv/validator/committee_queue.go (hunk 5 above)
Comment (AKorpusenko) [reply protocol/v2/ssv/validator/committee_queue.go]: @y0sher, fixed this. Added setting the `state.Slot` to `StartDuty` and  `HandleMessage`.

**Review by y0sher — APPROVED** [review]
looks ok! going to conflict with #1436  for sure though so proceed slowly :)

**Pull request merged**